### PR TITLE
SONARJAVA-387 Do not raise issue when enclosing method or class is de…

### PR DESCRIPTION
**Pull request by benzonico**
…precated

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java [hunk 1]
```diff
@@ -41,22 +44,54 @@
...
+  }
+
+  @Override
+  public void leaveNode(Tree tree) {
+    if (isDeprecatedMethod(tree) || isDeprecatedClassTree(tree)) {
+      nestedDeprecationLevel--;
+    }
+  }
+
+  private boolean isDeprecatedClassTree(Tree tree) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java]: MAJOR Make "isDeprecatedClassTree" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java [hunk 2]
```diff
@@ -41,22 +44,54 @@
...
+    return tree.is(Tree.Kind.METHOD, Tree.Kind.CONSTRUCTOR) && ((MethodTree) tree).symbol().isDeprecated();
+  }
+
+  private boolean isDeprecatedMethod(Tree tree) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java]: MAJOR Make "isDeprecatedMethod" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

File: java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java [hunk 3]
```diff
@@ -41,22 +44,54 @@
...
+        reportIssue(tree, "Remove this use of \"" + name + "\"; it is deprecated.");
       }
-      reportIssue(tree, "Remove this use of \"" + name + "\"; it is deprecated.");
+    } else if (isDeprecatedMethod(tree) || isDeprecatedClassTree(tree)) {
+      nestedDeprecationLevel++;
     }
+
+  }
+
+  private boolean isConstructor(Symbol symbol) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java]: MAJOR Make "isConstructor" a "static" method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java (hunk 1 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java]: should be called ``isDeprecatedClassTree``.

File: java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java (hunk 2 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java]: should be called ``isDeprecatedClassTree``.

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Pull request merged**